send_stream.go

```diff
@@ -379,14 +379,20 @@ func (s *sendStream) Close() error {
 		return nil
 	}
 	s.finishedWriting = true
-	if s.cancelWriteErr != nil {
-		s.mutex.Unlock()
-		return fmt.Errorf("close called for canceled stream %d", s.streamID)
-	}
-	s.ctxCancel(nil)
+	s.cancellationFlagged = true
+	cancelWriteErr := s.cancelWriteErr
```

Comment:
I think it's better to only set `s.cancellationFlagged = true` if cancelWriteErr was not true. I don't think this leads to a bug anywhere. 
Something like: 

```
cancelWriteErr := s.cancelWriteErr
if cancelWriteErr != nil {
    s.cancellationFlagged = true
}
```

Reply:
Done.